homeassistant/components/switcher_kis/quality_scale.yaml

```diff
@@ -0,0 +1,78 @@
+rules:
+  # Bronze
+  action-setup:
+    status: exempt
+    comment: The integration uses entity services.
+  appropriate-polling:
+    status: exempt
+    comment: The integration does not poll.
+  brands: done
+  common-modules: done
+  config-flow-test-coverage:
+    status: todo
+    comment: make sure flows end with created entry or abort
+  config-flow: done
+  dependency-transparency: done
+  docs-actions: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: todo
+  entity-event-setup: done
+  entity-unique-id: done
+  has-entity-name: done
+  runtime-data: todo
+  test-before-configure: done
+  test-before-setup:
+    status: exempt
+    comment: devices are setup asynchronously and marked as unavailable until they are ready.
+  unique-config-entry:
+    status: exempt
+    comment: The integration only supports a single config entry.
+
+  # Silver
+  action-exceptions: todo
+  config-entry-unloading: done
+  docs-configuration-parameters: done
+  docs-installation-parameters: done
+  entity-unavailable: done
+  integration-owner: done
+  log-when-unavailable: done
+  parallel-updates: todo
+  reauthentication-flow: done
+  test-coverage: done
+
+  # Gold
+  devices: done
+  diagnostics: done
+  discovery-update-info: done
+  discovery:
+    status: exempt
+    comment: There is no option to discover devices without adding the integration.
```

Comment:
Why not by the way?

Reply:
Background in https://github.com/home-assistant/architecture/discussions/970 https://github.com/home-assistant/architecture/discussions/970#discussioncomment-7119998

Switcher discovery uses UDP, currently Home Assistant only supports standard discovery methods (e.g. mDNS, DHCP, SSDP), the architecture discussion above tried to standardize this. Basically if HA would allow the integration to run any hook at startup it could detect devices and ask the user to confirm adding them.

Reply:
But can we use this to at least just detect a device on the network based on the mac address and then propose the user to add the integration and find stuff using UDP?

Reply:
Here is an example of the first 3 octets from 2 switcher devices I have:
```
cc:50:e3
e8:db:84
```
They are both registered to `Espressif Inc.`, moreover I have Shelly devices with the same `e8:db:84` prefix
I also considered DHCP but they use `ESP_` as prefix which is common to other vendors based on ESP SoC.
What do you suggest?

Reply:
Is their hostname representing the device? Or is it just the ESP hostname. If that is indeed the case, then this is it and we can't do discovery (assuming you checked zeroconf, for which there are nice tools coming in the beta today)

Reply:
> Is their hostname representing the device? Or is it just the ESP hostname. If that is indeed the case, then this is it and we can't do discovery.

It is the ESP hostname, they don't have any other discovery method, I have also discussed this with the vendor in the past but it didn't get anywhere (they try to help, but their firmware development is very slow to not existing).

> (assuming you checked zeroconf, for which there are nice tools coming in the beta today)

I run a full nightly instance (HA OS), yes this is new tools are great, all of the devices found under SSDP and Zeroconf  except for one already have core integration.